web/src/settings_components.js

```diff
@@ -240,7 +240,11 @@ function get_message_retention_setting_value($input_elem, for_api_data = true) {
     if ($custom_input.val().length === 0) {
         return settings_config.retain_message_forever;
     }
-    return Number.parseInt(Number($custom_input.val()), 10);
+    if (Number.isNaN(Number($custom_input.val()))) {
+        return Number.NaN;
+    }
+
+    return Number.parseInt($custom_input.val(), 10);
```

Comment:
This block would probably benefit from `const value = $custom_input.val()` rather than reading it from the DOM 3 times.